ovn-tester/ovn_workload.py

```diff
@@ -231,38 +251,73 @@ def provision(self, cluster):
         )
 
         # Route for traffic entering the cluster.
-        rp_gw = netaddr.IPAddress(self.gw_net.last - 1)
-        cluster.nbctl.route_add(self.gw_router, cluster.net, str(rp_gw))
+        rp_gw = self.gw_net.get_n4_ip_reverse()
+        if rp_gw and cluster.net.get_n4():
+            cluster.nbctl.route_add(self.gw_router,
+                                    network=cluster.net.get_n4(),
+                                    gw=str(rp_gw))
+        rp_gw6 = self.gw_net.get_n6_ip_reverse()
+        if rp_gw6 and cluster.net.get_n6():
+            cluster.nbctl.route_add(self.gw_router,
+                                    network=cluster.net.get_n6(),
+                                    gw=str(rp_gw6))
 
         # Default route to get out of cluster via physnet.
-        gr_def_gw = netaddr.IPAddress(self.ext_net.last - 2)
-        cluster.nbctl.route_add(self.gw_router, gw=str(gr_def_gw))
+        gr_def_gw = self.ext_net.get_n4_ip_reverse(2)
+        if gr_def_gw:
+            cluster.nbctl.route_add(self.gw_router, gw=str(gr_def_gw))
+        gr_def_gw6 = self.ext_net.get_n6_ip_reverse(2)
+        if gr_def_gw6:
+            cluster.nbctl.route_add(self.gw_router, network="::/0",
+                                    gw=str(gr_def_gw6))
 
         # Force return traffic to return on the same node.
-        cluster.nbctl.run(f'set Logical_Router {self.gw_router.name} '
-                          f'options:lb_force_snat_ip={gr_gw}')
+        if gr_gw:
+            cluster.nbctl.run(f'set Logical_Router {self.gw_router.name} '
+                              f'options:lb_force_snat_ip={gr_gw}')
+        if gr_gw6:
+            cluster.nbctl.run(f'set Logical_Router {self.gw_router.name} '
```

Comment:
This is not ok as it will overwrite the ipv4 setting above.

Reply:
@dceara ack. IIUC it seems we can't set it for both IPv4 and IPv6 at the same time.

Reply:
> @dceara ack. IIUC it seems we can't set it for both IPv4 and IPv6 at the same time.

It should be possible to.  The man page says:
[This option may have exactly one IPv4 and/or one IPv6 address on it, separated by a a space.](https://github.com/ovn-org/ovn/blob/93514df0d4c8fe7986dc5f287d7011f420d1be6d/ovn-nb.xml#L2291)
And we use it in a [system test](https://github.com/ovn-org/ovn/blob/93514df0d4c8fe7986dc5f287d7011f420d1be6d/tests/system-ovn.at#L3031) too.